packages/shell-api/src/database.ts

```diff
@@ -159,6 +159,31 @@ export default class Database extends ShellApiWithMongoClass {
     ) || [];
   }
 
+  _isFLE2Collection(collections: Document[], index: number): boolean {
+    return (
+      !collections[index].name.startsWith('enxcol_.') &&
+      collections.some(coll => coll.name === `enxcol_.${collections[index].name}.esc`) &&
+      collections.some(coll => coll.name === `enxcol_.${collections[index].name}.ecc`) &&
+      collections.some(coll => coll.name === `enxcol_.${collections[index].name}.ecoc`)
+    );
+  }
+
+  _getBadge(collections: Document[], index: number): string {
+    if (collections[index].type === 'timeseries') {
+      return '[time-series]';
+    }
+
+    if (collections[index].type === 'view') {
+      return '[view]';
+    }
+
+    if (this._isFLE2Collection(collections, index)) {
+      return '[rich-encrypted-query]';
+    }
+
+    return '';
+  }
```

Comment:
Would it make sense to move these two functions to `helpers.ts`? That’s where we typically put methods that are helpers and don’t rely on the object on which they are defined (i.e. don’t need `this`) :slightly_smiling_face: